roles/network_plugin/multus/tasks/main.yml

```diff
@@ -11,6 +11,12 @@
     - {name: multus-clusterrolebinding, file: multus-clusterrolebinding.yml, type: clusterrolebinding}
   register: multus_manifest_1
 
+# Caclico supports CNI more then 0.3.1 only after v3.23.0 https://github.com/projectcalico/calico/pull/5466
+- name: check for calico version
+  set_fact:
+    multus_cni_version: "0.3.1"
+  when: calico_version < "v3.23.0" and kube_network_plugin == "calico"
```

Comment:
We can specify `multus_cns_version` on an inventory file.
Automatic version selection could make confusion easily in long-term maintenance.

How about adding the version check into `roles/network_plugin/calico/tasks/check.yml` instead?
That would be helpful to let users know what was wrong.

Reply:
The idea was exactly not to make user trace calico version updates. I mean that when calico version in kubespray will be up to 3.23.0, user can benefit from more recent CNI version (0.4.0 or even 1.0.0). But if `multus_cni_version` will be specified in inventory - there is high probability, that it will stay at `0.3.1` even after calico version update.
But if reviewers will be against such PR, then I'll make version check in `roles/network_plugin/calico/tasks/check.yml`.

Reply:
@cristicalin  corrected by yours suggestions.
or I have to discard changes in `roles/network_plugin/multus/tasks/main.yml` and made them in `roles/network_plugin/multus/templates/multus-daemonset.yml.j2` ?

Reply:
For me it's ok like this.